Added formatting support

whiskeysierra opened the pull request:
Alternative implementation for #15

jhorstmann commented on src/main/java/org/zalando/jackson/datatype/money/MonetaryAmountSerializer.java:
```diff
@@ -24,21 +24,34 @@
 
 public final class MonetaryAmountSerializer extends JsonSerializer<MonetaryAmount> {
 
+    private final MonetaryAmountFormat format;
```
The source for `MonetaryAmountFormat` at https://github.com/JavaMoney/jsr354-api/blob/master/src/main/java/javax/money/format/MonetaryAmountFormat.java#L55 says

> Instances of this class are not required to be thread-safe

Addditionally, jackson has some support for configuring locales for serialization (https://fasterxml.github.io/jackson-databind/javadoc/2.2.0/com/fasterxml/jackson/databind/SerializationConfig.html#with%28java.util.Locale%29), which might be nice to reuse instead of having one single locale for the singleton ObjectMapper. I don't see yet how this config might be acessible from a `JsonSerializer`.

whiskeysierra replied:
> Instances of this class are not required to be thread-safe

Good catch!

> instead of having one single locale for the singleton ObjectMapper

I think the Locale that one can configure in Jackson's SerializationConfig is exactly that, a *single locale for the singleton ObjectMapper*.

whiskeysierra replied:
> I think the Locale that one can configure in Jackson's SerializationConfig is exactly that, a single locale for the singleton ObjectMapper.

I double checked and it looks like one can override it on a per-run basis:

```java
mapper.writer().with(locale).writeValueAsString(money);
```

whiskeysierra replied:
> I don't see yet how this config might be acessible from a JsonSerializer.

You can do the following on the given provider:

```java
final Locale locale = provider.getConfig().getLocale();
```

ptahchiev commented:
+1 on this one. Is there anything we can do to get it merged?

whiskeysierra commented:
> +1 on this one. Is there anything we can do to get it merged?

As @jhorstmann correctly pointed out, using the `MonetaryAmountFormat` as a singleton is potentially dangerous. We need to resolve this first.

whiskeysierra commented:
@jhorstmann @ptahchiev I updated the pull-request. It now:

- uses a factory to create format instances as needed (not sure how expensive this is, but at least it's threadsafe)
- new factory interface to create formats for a given locale
- provided implementation based on `MonetaryFormats.getAmountFormat(Locale, String...)`
- added formatting section to readme

jhorstmann commented:
:+1: Looks good. Would be cool if the money api already included the required factory interface, but having it here with a reasonable default is also fine.

whiskeysierra commented:
@zalando/pennybags if there are no objections I'd merge and release this one.
